geonode/utils.py

```diff
@@ -1963,7 +1963,26 @@ def get_supported_datasets_file_types():
             supported_types[default_types_id.index(_type.get("id"))] = _type
         else:
             supported_types.extend([_type])
-    return supported_types
+
+    # Order the formats (to support their visualization)
+    formats_order = ["vector", "raster", "archive", "other"]
+    types_ordered = {"vector": [], "raster": [], "archive": [], "other": []}
+    for resource_type in supported_types:
+        format = resource_type.get("format")
+        if format:
+            ordered_format = format if format in types_ordered.keys() else None
+            if ordered_format:
+                types_ordered[format].append(resource_type)
+            else:
+                types_ordered["other"].append(resource_type)
+        else:
+            types_ordered["other"].append(resource_type)
+
+    # Flatten the lit
+    supported_types_ordered = []
```

Comment:
@giohappy i'll evaluate to keep it as minimal as possible. 

```python
formats_order = [("vector", 0), ("raster",1), ("archive",2), ("other",3)]

ordered_payload = (
    (weight[1], resource_type )
    for resource_type  in supported_types
    for weight in formats_order
    if resource_type.get("format") in weight[0]
)

sorted_layers = [x[1] for x in sorted(ordered_payload, key=lambda x: x[0])]
```

In this way, the first line with the `formats_order` can also be used as config and let the user decide the visualization order